packages/gigamon/data_stream/ami/elasticsearch/ingest_pipeline/default.yml

```
@@ -913,6 +913,36 @@ processors:
       tag: set_dns_question_name
       copy_from: gigamon.ami.dns_query
       if: ctx.gigamon?.ami?.dns_query != null
+  - script:
+      lang: painless
+      description: Populates the dns.question.top_level_domain value
+      source: >-
+        String[] parts = ctx.dns.question.name.splitOnToken(".");
+        if (parts.length > 0) {
+          ctx.dns.question.top_level_domain = parts[parts.length - 1];
+        }
```

```suggestion
        int idx = ctx.dns.question.name.lastIndexOf('.');
        if (idx < 0) {
          return;
        }
        ctx.dns.question.top_level_domain = ctx.dns.question.name.substring(idx+1);
```
Again, no string allocations in the unhappy case and only one in the happy case.

---

Changes made as per the suggestion